components/authentication/authentication.js

```diff
@@ -414,17 +416,81 @@ function generatePasswordRecoveryURL (token) {
     return `${process.env.DEV_APP_DOMAIN}/password-reset/recover?token=${token}`
 }
 
+async function triggerVerificationCodeEmail(req, res) {
+    // Do logic. Trigger verification code e-mail as necessary
+    const { email } = req.body;
+    try {
+        const emailVerifiedValue = await isEmailVerified(email);
+
+        if (emailVerifiedValue) return res.status(400).send({ err: 'E-mail address is already verified'});
+
+        const verificationData = await createVerificationData(email);
+        if (!verificationData) return res.status(400).send({
+            err: 'This is an invalid request for this e-mail'
+        });
+
+        if (!verificationData.code) {
+            throw new Error("Verification code is undefined");
+        }
+        await webHook.sendEmail("/verification-code", { recipient: email, code: verificationData.code })

```

Comment:
This can be made to be fire and forget, like just return an "OK" and "check your e-mail within a few minutes". It may take a few hours for the email to bounce back in case the email is wrong.